Modify nsPluginInstanceOwner::GetPluginPortFromWidget so it returns the NSView
ChildWindow object in the window field for SciMoz plugins. This lets us set the
ScintillaView NSView object as a subview of the ChildWindow view.
diff -r 6db116c397b1 dom/plugins/base/nsPluginInstanceOwner.cpp
--- a/dom/plugins/base/nsPluginInstanceOwner.cpp	Mon Sep 17 20:00:29 2012 -0500
+++ b/dom/plugins/base/nsPluginInstanceOwner.cpp	Tue Oct 02 13:56:14 2012 -0700
@@ -2432,12 +2460,24 @@
 #ifdef XP_WIN
     if (mPluginWindow && (mPluginWindow->type == NPWindowTypeDrawable))
       result = mWidget->GetNativeData(NS_NATIVE_GRAPHIC);
     else
 #endif
 #ifdef XP_MACOSX
+    //ACTIVESTATE: Check the mimeType
+    bool AS_setPluginWindowWindowToChildView = false;
+    if (mInstance) {
+      const char*       mimeType;
+      mInstance->GetMIMEType(&mimeType);
+      static const char sciMozMimeType[] = "application/x-scimoz-plugin";
+      AS_setPluginWindowWindowToChildView = !PL_strcasecmp(mimeType, sciMozMimeType);
+    }
+    if (AS_setPluginWindowWindowToChildView) {
+      mPluginWindow->window = mWidget->GetNativeData(NS_NATIVE_WIDGET);
+    } else // no-brace: do regular code
+    // END ACTIVESTATE: Check the mimeType
     if (GetDrawingModel() == NPDrawingModelCoreGraphics ||
         GetDrawingModel() == NPDrawingModelCoreAnimation ||
         GetDrawingModel() == NPDrawingModelInvalidatingCoreAnimation)
       result = mWidget->GetNativeData(NS_NATIVE_PLUGIN_PORT_CG);
     else
 #endif
